lib/TfIdfVectorizer.jsm

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+// Javascript doesn't support unicode page mnemonics out of the box, and so
+// we have to define them ourselves.
+const UNICODE_SPACE = "\x20\xA0\u1680\u2000-\u200A\u2028\u2029\u202F\u205F\u3000";
```

Comment:
All these constants here want some comments to clarify:
* What they are in general
* Some extra reference links, which pointing to more details about those unicodes, would be even better

Reply:
Unicode specifies certain mnemonics for code pages and character classes. They call them "character properties" https://en.wikipedia.org/wiki/Unicode_character_property . These mnemonics are have been adopted by many regular expression libraries, however the standard regexp system doesn't support unicode character properties. If you want to use them, you have to pull in a library like XRegExp. 

Each of these sections contains the characters values / ranges for specific character property: Whitespace, Symbol (S), Punctuation (P), Number (N), Mark (M), and Letter (L). The tables are defined by Unicode. I think I reformatted the data from XRegExp to get it to work here.

We have to support unicode the tokenizer, as we expect to see (and have seen in our training data) nonenglish and multilingual documents. This allows us to use the same code for any language that uses spaces to separate words.

Interestingly enough, it looks like xregexp is pulled in via two or three levels of dependencies, but I have no idea how / should you'd import it directly.

Reply:
Could you add this to the code?